---
title: "Class16"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


##The data for for hands-on session comes from GEO entry: GSE37704, which is associated with the following publication:

Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703
The authors report on differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1.


##Load in DESeq2

```{r}
library(DESeq2)
```

## load in .CSV files (countDATA and MEtadata)

```{r}

metaFile<-read.csv("GSE37704_metadata.csv")
countFile<-read.csv("GSE37704_featurecounts.csv")
```
```{r}
# Import metadata and take a peak
colData = read.csv("GSE37704_metadata.csv", row.names=1)
head(colData)

# Import countdata
rawcountData = read.csv("GSE37704_featurecounts.csv", row.names=1)
head(rawcountData)
```

```{r}
countData<-rawcountData[,-1]
head(countData)
```

```{r}
dim(countData)
```
There are genes that have zero counts in all samples (i.e. columns). We should remove these.

First Identify Zero sum rows
```{r}
#rowSums(countData gives a lil numeric with all the rows sums)

rowSums(countData) == 0
#make new variable that only re
countData <-countData[rowSums(countData)!=0, ]
```

```{r}
dim(countData)
```
#running DESeq

```{r}
dds = DESeqDataSetFromMatrix(countData=countData,
                             colData=colData,
                             design=~condition)
dds = DESeq(dds)
```

````{r}

res <- results(dds)
head(res)
```


```{r}
plot(res$log2FoldChange, -log(res$padj))
```
#Lets Add Some Color

```{r}
mycols<- rep("gray", nrow(res))
mycols[abs(res$log2FoldChange) > 2] <- "blue"
plot(res$log2FoldChange, -log(res$padj), col=mycols)
```

To add Annotation to my Gene list, I will install some bioconductor packages. However, I will do this in the console and not in the Rmd file!

```{r}

# BiocManager::install("AnnotationDbi")
# BiocManager::install("Org")
```

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
columns(org.Hs.eg.db)
```

Let's map our ENSEMBLE gene ids to the more conventional gene symbol

```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                    keys=row.names(res), 
                    keytype="ENSEMBL",
                    column="SYMBOL",
                    multiVals="first")

res$entrez <- mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")

res$name <-   mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="GENENAME",
                    multiVals="first")

head(res, 10)

head(res)
```

```{r}
write.csv(res, file="DESeq_Results.csv")
```

# Pathway analysis

Here we are going to use the **gage** package for pathway analysis. Once we have a list of enriched pathways, we're going to use the **pathview** package to draw pathway diagrams, shading the molecules in the pathway by their degree of up/down-regulation.
```{r}
# Run in your R console (i.e. not your Rmarkdown doc!)
# BiocManager::install( c("pathview", "gage", "gageData") )
```

#Load these Packages

```{r}

library(gage)
library(gageData)
library(pathview)
```
```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]

# Examine the first 3 pathways
head(kegg.sets.hs, 3)
```

Our input will be a vector of fold change values with ENTREZ gene ids as names

```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```

```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
attributes(keggres)

keggres$less
```

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```
```{r}
knitr::include_graphics("hsa04110.pathview.png")
```

